src/main/java/io/v4guard/plugin/bungee/listener/AntiVPNListener.java

```diff
@@ -10,6 +10,7 @@ public class AntiVPNListener implements Listener {
 
     @EventHandler(priority = Byte.MIN_VALUE)
     public void onPreLogin(PreLoginEvent e) {
+        if(e.getConnection() == null) return;
```

Comment:
This check avoids verifying null connections, which would be an unnecessary waste of bandwidth and would count as a request to the api.